src/Binding/JSWebSocketClient.cpp

```diff
@@ -251,43 +205,35 @@ nidium_WebSocket_constructor(JSContext *cx, unsigned argc, JS::Value *vp)
         free(path);
         free(host);
         free(durl);
-        return false;
+        return nullptr;
     }
 
-    JSWebSocket *wss = new JSWebSocket(ret, cx, host, port, path, isSSL);
+    JSWebSocket *wss = new JSWebSocket(host, port, path, isSSL);
 
     free(path);
     free(host);
     free(durl);
 
-    JS_SetPrivate(ret, wss);
-
-    args.rval().setObjectOrNull(ret);
-
-    NidiumJSObj(cx)->rootObjectUntilShutdown(ret);
```

Comment:
Also `NidiumJSObj(cx)->unrootObject(obj);` needs to be replaced by `this->unroot()`